### PR TITLE
Add `m.fragment()`

### PR DESCRIPTION
**Pull request by pygy**
As announced in #1241, I've also added `./hyperscript`, which combines `render/hyperscript`, `render.raw` and `render.fragment`.

### CODE REVIEW

File: render/fragment.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+"use strict"
+
+var Vnode = require("../render/vnode")
+
+module.exports = function() {
+        var attrs, children, childrenIndex = 0
```
Comment (tivac) [inline render/fragment.js]: nitpicking but indentation on this line is weird

**Comment by tivac** [comment]
LGTM

File: render/fragment.js (hunk 1 above)
Comment (pygy) [inline render/fragment.js]: My bad, my editor defaults to spaces. Updating (and adding a couple of tests).

Comment (pygy) [inline render/fragment.js]: Fixed (I hadn't seen the double indentation there).

**Comment by pygy** [comment]
In ES6, it would have been tempting to implement it as 
```
function fragment(...args) {
    return hyperscript(fragment, ...args)
}
```
and convert `fragment` to `'['` in `hyperscript. Doing that in ES5 is quite ugly and inefficient though, so I copied/pasted the relevant section from `hyperscript` instead.

**Comment by pygy** [comment]
Thinking about this a bit more, what about `m(m.fragment [, attrs] [, children...])` where `m.fragment` is just a sentinel object attached to `hyperscript`? That way we could avoid duplicating the `hyperscript` logic.

**Comment by lhorie** [comment]
I'm thinking the signature should be simpler, like `m.fragment(attrs, children)` (no optional args). The overload `fragment(attrs)` seems pointless, and `fragment(children)` is generally going to be the same thing as `children`

**Comment by barneycarroll** [comment]
[This is a really good point](https://github.com/lhorie/mithril.js/pull/1249#issuecomment-240619063): a new fragment hyperscript signature only makes sense in the case of having both `attrs, children`, at which point it is simply the default hyperscript signature without the leading argument. Are we dead set against that? `m( attrs, children )`

**Comment by pygy** [comment]
So I chopped off a good deal out of it.

**Pull request merged**